src/Compilers/Core/Portable/SourceGeneration/ISyntaxReceiver.cs

```diff
@@ -22,8 +33,11 @@ namespace Microsoft.CodeAnalysis
     /// 
     /// A new instance of <see cref="ISyntaxReceiver"/> is created per-generation, meaning the instance
     /// is free to store state without worrying about lifetime or reuse.
+    /// 
+    /// An <see cref="ISourceGenerator"/> may provide only a single <see cref="ISyntaxReceiver"/> or
+    /// <see cref="ISyntaxReceiverWithContext" />, not both.
     /// </remarks>
-    public interface ISyntaxReceiver
+    public interface ISyntaxReceiver : ISyntaxReceiverBase
```

Comment:
can we obsolete this and push people to use ISyntaxReceiverWithContext ?

Reply:
Yep, we can definitely take that route.

Reply:
I would also push for obsoleting this API, but we do need compat council approval. Please send an email.